flink-ml-iteration/flink-ml-iteration-common/src/main/java/org/apache/flink/iteration/datacache/nonkeyed/ListStateWithCache.java

```diff
@@ -82,11 +82,13 @@ public ListStateWithCache(
                                 runtimeContext.getUserCodeClassLoader());
         if (fraction > 0) {
             MemoryManager memoryManager = containingTask.getEnvironment().getMemoryManager();
+            // TODO Adds a configuration to set the fraction of the memory that the
+            // ListStateWithCache can use.
             segmentPool =
                     new LazyMemorySegmentPool(
                             containingTask,
                             memoryManager,
-                            memoryManager.computeNumberOfPages(fraction));
+                            memoryManager.computeNumberOfPages(fraction) / 2);
```

Comment:
That's a good point. I also think that the `iteration.memory.feedback-buffer.fraction` we just added is inappropriate because it is not defined in `ManagedMemoryUseCase` of Flink. So it would grab the managed memory with `ListStateWithCache` which may cause memory allocation failed. 

I changed the implementation to also use the `ManagedMemoryUseCase.OPERATOR` to share the managed memory with `ListStateWithCache`. Would you have another look?

Reply:
The changes are acceptable to me. Thank you for your efforts in improving them!